src/daft-dsl/src/expr.rs

```diff
@@ -190,9 +190,7 @@ impl AggExpr {
                     ))),
                 }
             }
-            MapGroups { func, inputs } => {
-                func.to_field(inputs.as_slice(), schema, &Expr::Agg(self.clone()))
```

Comment:
The `Expr::Agg` here is actually useless: `MapGroups` only allow UDFs, and `PythonUDF` doesn't use the `Expr` passed in indeed.
If we really want the `FunctionEvaluator` to know if we're in an aggregation, an additional `Context(Default/Agg)` argument would make more sense I think.

Reply:
Ah you're right! @colin-ho Was the rationale behind recreating the expression?